client/state/data-layer/wpcom/sites/stats/post/index.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	POST_STATS_REQUEST,
+	POST_STATS_REQUEST_START,
+	POST_STATS_REQUEST_SUCCESS,
+	POST_STATS_REQUEST_FAILURE
+} from 'state/action-types';
+import { receivePostStat } from 'state/stats/posts/actions';
+
+export const requestPostStat = ( { dispatch }, { stat, siteId, postId } ) => {
+	dispatch( {
+		type: POST_STATS_REQUEST_START,
+		stat,
+		postId,
+		siteId
+	} );
+
+	return wpcom.site( siteId ).statsPostViews( postId, {
+		fields: stat
+	} ).then( data => {
+		if ( stat in data ) {
+			dispatch( receivePostStat( stat, siteId, postId, data[ stat ] ) );
+		}
+		dispatch( {
+			type: POST_STATS_REQUEST_SUCCESS,
+			stat,
+			siteId,
+			postId
+		} );
+	} ).catch( error => {
+		dispatch( {
+			type: POST_STATS_REQUEST_FAILURE,
+			stat,
+			siteId,
+			postId,
+			error
+		} );
+	} );
+};
```

Comment:
Great idea! I pushed a first attempt here 4a5eb36

It's working correctly but the `batch` endpoint could be improved: For now, when an URL returns an error, the status_code is included in the response, but when it's successful, it's not. I think it should behave consistently and always include an "envelope" to the response.